haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import warnings
+from collections import defaultdict
+from typing import List, Union, Optional, Dict, Literal
+
+from dateutil.parser import parse, ParserError
+
+from haystack.nodes.ranker.base import BaseRanker
+from haystack.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class RecentnessRanker(BaseRanker):
+    outgoing_edges = 1
+
+    def __init__(
+        self,
+        date_identifier: str,
+        weight: float = 0.5,
+        top_k: Optional[int] = None,
+        method: Literal["reciprocal_rank_fusion", "score"] = "reciprocal_rank_fusion",
+    ):
+        """
+        This Node is used to rerank retrieved documents based on their age. Newer documents will rank higher.
+        The importance of recentness is parametrized through the weight parameter.
+
+        :param date_identifier: Identifier pointing to the date field in the metadata.
+                This is a required parameter, since we need dates for sorting.
+        :param weight: in range [0,1].
+                0 disables sorting by age.
+                0.5 content and age have the same impact
+                1 means sorting only by age, most recent comes first.
```

Comment:
Should be straightforward to implement, I can do it should the need arise. Off the top of my head, I couldn't come up with use cases where it would be very useful to have oldest documents prioritised in the search, so I agree that it's probably not worth including in the initial PR.